cylc/flow/network/schema.py

```diff
@@ -624,19 +621,18 @@ class Meta:
     string_extended = String()
 
 
-# The run mode for the workflow.
-WorkflowRunMode = graphene.Enum(
-    'WorkflowRunMode',
-    [(m.capitalize(), m) for m in WORKFLOW_RUN_MODES],
-    description=lambda x: RunMode(x.value).describe() if x else None,
-)
+class TaskRunMode(graphene.Enum):
+    """The mode used to run a task."""
 
-# The run mode for the task.
-TaskRunMode = graphene.Enum(
-    'TaskRunMode',
-    [(m.capitalize(), m) for m in TASK_CONFIG_RUN_MODES],
-    description=lambda x: RunMode(x.value).describe() if x else None,
-)
+    # NOTE: using a different enum because:
+    # * We only want to offer a subset of run modes (REQUEST_* only).
+
+    Live = cast('Enum', RunMode.LIVE)  # type: graphene.Enum
+    Skip = cast('Enum', RunMode.SKIP)  # type: graphene.Enum
+
+    @property
+    def description(self):
+        return RunMode(self.value).describe()
```

Comment:
I think this needs to be synced with [recent changes](https://github.com/cylc/cylc-flow/pull/6554).

Can we not generate these enums from their source definitions any more?

@wxtim, please could you track this.

Reply:
> Can we not generate these enums from their source definitions any more?

I think I was just trying to make it consistent with `StopMode` .. Which does the same thing (in just using a subset of stop modes)..

Also, it was throwing an error for some reason.. But if it's a big deal I can try use the former

Reply:
I'm working through this.. with reference with this change, will have something up tomorrow

Reply:
I've reverted this .. And made the appropriate changes to work with @wxtim 's merged work

Reply:
@wxtim, please review this and resolve this comment when happy.